pkg/operator/ceph/cluster/rbd/controller.go

```diff
@@ -284,29 +313,42 @@ func (r *ReconcileCephRBDMirror) reconcileCreateCephRBDMirror(cephRBDMirror *cep
 }
 
 // updateStatus updates an object with a given status
-func (r *ReconcileCephRBDMirror) updateStatus(observedGeneration int64, name types.NamespacedName, status string) {
+func (r *ReconcileCephRBDMirror) updateStatus(observedGeneration int64, name types.NamespacedName, status string, cephx *cephv1.CephxStatus) error {
 	rbdMirror := &cephv1.CephRBDMirror{}
-	err := r.client.Get(r.opManagerContext, name, rbdMirror)
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			logger.Debug("CephRBDMirror resource not found. Ignoring since object must be deleted.")
-			return
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		err := r.client.Get(r.opManagerContext, name, rbdMirror)
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				logger.Debug("CephRBDMirror resource not found. Ignoring since object must be deleted.")
+				return nil
+			}
+			return errors.Wrapf(err, "failed to retrieve CephRBDMirror %q to update status to %q", name, status)
 		}
-		logger.Warningf("failed to retrieve rbd mirror %q to update status to %q. %v", name, status, err)
-		return
-	}
 
-	if rbdMirror.Status == nil {
-		rbdMirror.Status = &cephv1.Status{}
-	}
+		if rbdMirror.Status == nil {
+			rbdMirror.Status = &cephv1.Status{}
+		}
 
-	rbdMirror.Status.Phase = status
-	if observedGeneration != k8sutil.ObservedGenerationNotAvailable {
-		rbdMirror.Status.ObservedGeneration = observedGeneration
-	}
-	if err := reporting.UpdateStatus(r.client, rbdMirror); err != nil {
-		logger.Errorf("failed to set rbd mirror %q status to %q. %v", rbdMirror.Name, status, err)
-		return
+		rbdMirror.Status.Phase = status
+		if observedGeneration != k8sutil.ObservedGenerationNotAvailable {
+			rbdMirror.Status.ObservedGeneration = observedGeneration
+		}
+
+		if cephx != nil {
+			rbdMirror.Status.Cephx.Daemon = *cephx
+		}
+
+		if err := reporting.UpdateStatus(r.client, rbdMirror); err != nil {
+			return errors.Wrapf(err, "failed to set CephRBDMirror %q status to %+v", name, rbdMirror.Status)
```

Comment:
log the status and lets not return it?

Reply:
I meant to respond to this yesterday. We need to return an error here. It is critical that we don't lose status updates that update cephx key metadata so that Rook doesn't unexpectedly rotate keys multiple times after a rotation is initially completed. We have been updating all Rook controllers that have cephx statuses to return errors when the status update fails because of this